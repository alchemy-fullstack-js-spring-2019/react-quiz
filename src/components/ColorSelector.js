import React from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ changeHandler }){
  return (
      <>
      <input name="color" type="color" onChange={changeHandler}/>
      <input name="backgroundColor" type="color" onChange={changeHandler}/>
      </>
  );
}


ColorSelector.propTypes = {
  changeHandler: PropTypes.func.isRequired
};

export default ColorSelector;
