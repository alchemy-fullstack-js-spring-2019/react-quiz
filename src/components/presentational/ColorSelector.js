import React from 'react';
import PropTypes from 'prop-types';

export default function ColorSelector({ color, handleChange }) {
  return (
    <input name="color" value={color} type="color" onChange={handleChange} />
  );
}
  
ColorSelector.propTypes = {
  color: PropTypes.string,
  handleChange: PropTypes.func
};
