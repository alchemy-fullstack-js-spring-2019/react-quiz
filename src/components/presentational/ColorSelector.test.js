import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('ColorSelector Test', () => {
  it('renders ColorSelector', () => {
    const wrapper = shallow(<ColorSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
