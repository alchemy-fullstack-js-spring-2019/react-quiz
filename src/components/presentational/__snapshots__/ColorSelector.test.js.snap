// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector Test renders ColorSelector 1`] = `
<input
  name="color"
  type="color"
/>
`;
